extensions/objects/templates/math-expression-content-editor.component.ts

```diff
@@ -176,12 +183,14 @@ export class MathExpressionContentEditorComponent implements OnInit {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes.value &&
-      changes.value.currentValue.raw_latex !==
-      changes.value.previousValue.raw_latex) {
+    let value = changes.value;
+    if (value &&
+      (value.currentValue.raw_latex !== value.previousValue.raw_latex ||
+        this.localValue.label !== value.currentValue.raw_latex)) {
```

Comment:
Why this change? Please explain.

Reply:
I realise now that the current code is a bit hard to understand to readers -- thanks for pointing it out. I have updated the code and made it a bit simpler now. PTAL

Reply:
Thanks -- I at least understand it now! But will defer to @iamprayush for the actual review.